examples/ipv6only-app/esp32/main/main.cpp

```diff
@@ -0,0 +1,650 @@
+
+    result->error = chip_rpc_CONNECTION_ERROR_OK;
+    return pw::OkStatus();
+}
+
+void UdpReceiver(void * pvParameters)
+{
+    int portno;                     // port to listen on
+    struct sockaddr_in6 serveraddr; // server's addr
+    char buf[kUdpBufferSize];       // rx message buf
+    char * hostaddrp;               // dotted decimal host addr string
+    int optval;                     // flag value for setsockopt
+    int n;                          // message byte size
+
+    while (1)
+    {
+
+        // Wait for wifi be initialized and ip6 address assigned.
+        xSemaphoreTake(wifi_init_semaphore, portMAX_DELAY);
+
+        ESP_LOGI(TAG, "UDP server (re)starting");
+
+        portno = 8765;
+        // socket: create the parent socket
+        sockfd = socket(AF_INET6, SOCK_DGRAM, 0);
+        if (sockfd < 0)
+        {
+            ESP_LOGE(TAG, "ERROR opening socket");
+            assert(0);
+            return;
+        }
+
+        // setsockopt: Handy debugging trick that lets
+        // us rerun the server immediately after we kill it;
+        // otherwise we have to wait about 20 secs.
+        // Eliminates "ERROR on binding: Address already in use" error.
+        optval = 1;
+        setsockopt(sockfd, SOL_SOCKET, SO_REUSEADDR, static_cast<const void *>(&optval), sizeof(int));
+
+        // build the server's Internet address
+        memset(&serveraddr, 0, sizeof(serveraddr));
+        serveraddr.sin6_len    = sizeof(serveraddr);
+        serveraddr.sin6_family = AF_INET6;
+        serveraddr.sin6_addr   = in6addr_any;
+        serveraddr.sin6_port   = htons((unsigned short) portno);
+
+        //  bind: associate the parent socket with a port
+        if (bind(sockfd, reinterpret_cast<struct sockaddr *>(&serveraddr), sizeof(serveraddr)) < 0)
+        {
+            ESP_LOGE(TAG, "ERROR on binding");
+            assert(0);
+            return;
+        }
+
+        ESP_LOGI(TAG, "UDP server bound to port %d", portno);
+
+        // main loop: wait for a datagram, then respond
+        clientlen = sizeof(clientaddr);
+        wifi_ap_record_t ap_info;
+        fd_set readset;
+        while (ESP_OK == esp_wifi_sta_get_ap_info(&ap_info))
+        {
+            // recvfrom: receive a UDP datagram from a client
+            memset(buf, 0, sizeof(buf));
+
+            FD_ZERO(&readset);
+            FD_SET(sockfd, &readset);
+            // TODO: somehow this throws an exception on wifi disconnect
+            // "Guru Meditation Error: Core  0 panic'ed (LoadProhibited). Exception was unhandled"
+            int select_err = select(sockfd + 1, &readset, nullptr, nullptr, nullptr);
+            if (select_err < 0)
+                continue;
+
+            n = recvfrom(sockfd, buf, kUdpBufferSize, 0, reinterpret_cast<struct sockaddr *>(&clientaddr), &clientlen);
+            if (n < 0)
+                continue;
+            else if (strncmp(buf, "reboot", 3) == 0)
```

Comment:
This actually checks for `reb` reather than `reboot`. Is that intended?

Reply:
Removed